src/Tempest/Mapper/src/Mappers/ObjectToArrayMapper.php

```diff
@@ -16,11 +21,61 @@ public function canMap(mixed $from, mixed $to): bool
     }
 
     public function map(mixed $from, mixed $to): array
+    {
+        $properties = $this->resolveProperties($from);
+        $mappedProperties = [];
+
+        foreach ($properties as $property_name => $property_value) {
+            try {
+                $property = PropertyReflector::fromParts(class: $from, name: $property_name);
+                $property_name = $this->resolvePropertyName($property);
+                $property_value = $property_value;
+
+                // @TODO May be removed if we want to handle private/protected properties
+                if (! $property->isPublic()) {
+                    continue;
+                }
+
+                $mappedProperties[ $property_name ] = $property_value;
+            } catch (ReflectionException) {
+                continue;
+            }
+        }
+
+        return $mappedProperties;
+    }
+
+    /**
+     * @return array<string, mixed> The properties name and value
+     */
+    private function resolveProperties(object $from): array
     {
         if ($from instanceof JsonSerializable) {
             return $from->jsonSerialize();
         }
 
-        return (array) $from;
+        try {
+            $class = new ClassReflector($from);
+            $properties = $class->getProperties();
```

Comment:
You can use `getPublicProperties()` instead, but you can keep the changes in `PropertyReflector`, they make sense even though not needed for this feature